### PR TITLE
LPS-127081 [WIP] Add frontend-js-state-web

### PR DESCRIPTION
**Pull request by wincent**
~Just parking this here so I can pull it down onto another machine and try it there.~

~This is about the 4th variation of "trying to get DXP to load a JS file" and it seems like it "should" work, but I'm getting loader errors:~

```
liferay-amd-loader | Errors returned from server for require( ["@liferay/frontend-js-state-web@1.0.0/State"] ): ["Missing required module '@liferay/frontend-js-state-web@1.0.0/State'"]
```

~Ditto when I name the module without a leading `@liferay/` named scope. So I suspect some rubbish in the cache and if I am going to have to `ant all` a couple times to get this working, I'd rather do it on a faster machine (although... it might not save me any time because I bet I don't have the stuff on that machine to actually run tomcat, but we'll see).~

~For the record, the first thing I tried was a `Fragment-Host` mechanism, seeing as this is defined in `frontend-js-web` (ie. on `Liferay`) but built separately. And then injecting it into the page with `Liferay-JS-Resources-Top-Head`. That didn't work because the build output calls `Liferay.Loader.define` and throws an error because nobody is requesting the module yet. So, tried adding an `.npmbundlerrc` to get rid of the AMD wrapping, but that breaks TypeScript because Typescript _needs_ the `export` in there, and we don't want a naked `exports.default = blah` to wind up in the JS; that would just break the page. Then I started in on other options inspired by `git grep TopHead` before finally settling on the approach seen in this draft because it's something that I've seen work before for `remote-app-support-web`.~

**Update:** Got it working in the end. Root cause was a typo in the module path; forgot to update it as I went through the various iterations mentioned above 🤦‍♂️.

### CODE REVIEW

File: modules/apps/frontend-js/frontend-js-state-web/.babelrc.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+module.exports = {
```
Comment (jbalsas) [inline modules/apps/frontend-js/frontend-js-state-web/.babelrc.js:15]: So, my question from https://github.com/liferay/liferay-frontend-projects/pull/425#issuecomment-788724223

Why not using `webpack` directly using the `"mode": "everlasting-damaging"` that we recently coined?

Comment (wincent) [reply modules/apps/frontend-js/frontend-js-state-web/.babelrc.js:15]: I'll go and answer it over there.

Comment (jbalsas) [reply modules/apps/frontend-js/frontend-js-state-web/.babelrc.js:15]: Sounds good to me. Please, proceed.

**Comment by wincent** [comment]
Just rebased this (taking code as it existed a couple of days ago) and am confirming now that it works with `ant all` and [the new prerelease of `npm-scripts`](https://github.com/liferay/liferay-frontend-projects/releases/tag/npm-scripts%2Fv38.2.0-pre.0). I have some more stuff to add to this and will send that one to the `liferay-frontend` user soon; if not today, Monday. Closing this one for now.

**Pull request closed without merging**